server/datastore/mysql/hosts.go

```diff
@@ -2141,7 +2141,36 @@ func (ds *Datastore) getOrInsertMunkiIssues(ctx context.Context, errors, warning
 	return msgToID, nil
 }
 
-func (ds *Datastore) SetOrUpdateMDMData(ctx context.Context, hostID uint, isServer, enrolled bool, serverURL string, installedFromDep bool, name string) error {
+func (ds *Datastore) SetOrUpdateMDMData(
+	ctx context.Context,
+	hostID uint,
+	isServer, enrolled bool,
+	serverURL string,
+	installedFromDep bool,
+	name string,
+) error {
+	// MDM queries return an empty server URL when the host is not enrolled to an MDM.
+	if serverURL == "" {
```

Comment:
> UPDATE: I just read another PR and found out that apparently a host can be unenrolled but have a server-url.

Huh, thanks for catching this. I'll take another look.

Reply:
mmm if osquery cleans the server URL (which sounds like it does) then we can just adjust the code here: 

https://github.com/fleetdm/fleet/blob/f64ba83b2ea35e571cf4278c5db168921ffe3a25/server/datastore/mysql/apple_mdm.go#L542-L550

to do the same to be safe? This was an overlook on my side

Reply:
> mmm if osquery cleans the server URL (which sounds like it does) then we can just adjust the code here:

Am not following. 

Maybe you folks are talking about `enrolled` in `host_mdm`, but this is the `enrolled` field coming from query ingestion (macadmins table [mdm](https://github.com/macadmins/osquery-extension/blob/ade4391e23036c028f190a0f208a3a9d3715d519/tables/mdm/mdm.go#L80-L134)).

Looks like the [mdm macadmins extension](https://github.com/macadmins/osquery-extension/blob/ade4391e23036c028f190a0f208a3a9d3715d519/tables/mdm/mdm.go#L80-L134) won't return entries with `non-empty server_url AND enrolled = false` (only time it will return `enrolled = false` is when it's DEP capable and it's not enrolled, in such case `server_url` is returned empty AFAICS).

Let me know if I'm making sense.

Reply:
> Looks like the [mdm macadmins extension](https://github.com/macadmins/osquery-extension/blob/ade4391e23036c028f190a0f208a3a9d3715d519/tables/mdm/mdm.go#L80-L134) won't return entries with `non-empty server_url AND enrolled = false`

I think this is correct.

> (only time it will return `enrolled = false` is when it's DEP capable and it's not enrolled, in such case `server_url` is returned empty AFAICS)

Per my understanding, this is not correct, it will return `enrolled = false` if a payload o type `com.apple.mdm` doesn't exist OR it exists but doesn't contain any items (weird, but maybe possible?)

> Let me know if I'm making sense.

it makes sense to me, what do you think if I submit a PR to update `UpdateHostTablesOnMDMUnenroll` (the function I linked before) to remove the row from the table like we do here instead of setting `enrolled = false`?

Reply:
> Per my understanding, this is not correct, it will return enrolled = false if a payload o type com.apple.mdm doesn't exist OR it exists but doesn't contain any items (weird, but maybe possible?)

Right, seems like the extension lists all installed profiles and filters out the ones that are not enrollment profiles.

_"The Mobile Device Management (MDM) payload, a simple property list, is designated by the `com.apple.mdm` value in the PayloadType field."_ (From the MDM protocol reference document.)

> it makes sense to me, what do you think if I submit a PR to update UpdateHostTablesOnMDMUnenroll (the function I linked before) to remove the row from the table like we do here instead of setting enrolled = false?

Why do we need this change?
Is it an issue to have the row in `host_mdm` with `enrolled = false` (maybe it's useful to have it there, am guessing all queries that use this table check for `enrolled = true`).

Reply:
> Why do we need this change?
Is it an issue to have the row in host_mdm with enrolled = false (maybe it's useful to have it there, am guessing all queries that use this table check for enrolled = true).

Ah, maybe we want this to make it consistent I guess (with the functionality am adding here).

Can we assume that Fleet will eventually ingest this info and remove the entry here?

Reply:
yes, we could that too, I'm worried about a host being enrolled into MDM but not having fleetd installed. It's an edge case tho.

Reply:
jinx! sorry, I didn't see your comment before I sent mine!